internal/util/helpers_test.go

```diff
@@ -4,18 +4,758 @@
+		envToken      string
+		expectedError bool
+	}{
+		{
+			name:          "Valid GRPC connection to localhost with secure connection",
+			grpcHost:      "127.0.0.1",
+			grpcPort:      8090,
+			allowInsecure: false,
+			issuerUrl:     "http://localhost:8081",
+			clientId:      "minder-cli",
+			envToken:      "MINDER_AUTH_TOKEN",
+			expectedError: false,
+		},
+		{
+			name:          "Valid GRPC connection to localhost with insecure connection",
+			grpcHost:      "127.0.0.1",
+			grpcPort:      8090,
+			allowInsecure: true,
+			issuerUrl:     "http://localhost:8081",
+			clientId:      "minder-cli",
+			envToken:      "MINDER_AUTH_TOKEN",
+			expectedError: false,
+		},
+		{
+			name:          "Valid GRPC connection to localhost without passing MINDER_AUTH_TOKEN as an argument",
+			grpcHost:      "127.0.0.1",
+			grpcPort:      8090,
+			allowInsecure: false,
+			issuerUrl:     "http://localhost:8081",
+			clientId:      "minder-cli",
+			envToken:      "",
+			expectedError: false,
+		},
+		{
+			name:          "Valid GRPC connection to remote server with secure connection",
+			grpcHost:      "api.stacklok.com",
+			grpcPort:      443,
+			allowInsecure: false,
+			issuerUrl:     "https://auth.stacklok.com",
+			clientId:      "minder-cli",
+			envToken:      "MINDER_AUTH_TOKEN",
+			expectedError: false,
+		},
+		{
+			name:          "Valid GRPC connection to remote server with insecure connection",
+			grpcHost:      "api.stacklok.com",
+			grpcPort:      443,
+			allowInsecure: true,
+			issuerUrl:     "https://auth.stacklok.com",
+			clientId:      "minder-cli",
+			envToken:      "MINDER_AUTH_TOKEN",
+			expectedError: false,
+		},
+		{
+			name:          "Valid GRPC connection to remote server without passing MINDER_AUTH_TOKEN as an argument",
+			grpcHost:      "api.stacklok.com",
+			grpcPort:      443,
+			allowInsecure: true,
+			issuerUrl:     "https://auth.stacklok.com",
+			clientId:      "minder-cli",
+			envToken:      "",
+			expectedError: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			originalEnvToken := os.Getenv(util.MinderAuthTokenEnvVar)
+			err := os.Setenv(util.MinderAuthTokenEnvVar, tt.envToken)
+			// reset this the environment variable when complete.
+			defer os.Setenv(util.MinderAuthTokenEnvVar, originalEnvToken)
+			if err != nil {
+				t.Errorf("error setting MinderAuthTokenEnvVar: %v", err)
+			}
+			conn, err := util.GetGrpcConnection(tt.grpcHost, tt.grpcPort, tt.allowInsecure, tt.issuerUrl, tt.clientId)
+			if (err != nil) != tt.expectedError {
+				t.Errorf("expected error: %v, got: %v", tt.expectedError, err)
+			}
+			if conn != nil {
+				err = conn.Close()
+				if err != nil {
+					t.Errorf("error closing connection: %v", err)
+				}
+			}
+
+		})
+	}
+}
+
+// TestSaveCredentials tests the SaveCredentials function
+func TestSaveCredentials(t *testing.T) {
+	t.Parallel()
+	var envLock sync.Mutex
+	envLock.Lock()
+	defer envLock.Unlock()
```

Comment:
Declaring a new `sync.Mutex` here and then acquiring it doesn't do much, because the mutex isn't shared across multiple goroutines (e.g. `t.Run()` spawns a new goroutine, but this doesn't call `t.Run`).

Reply:
Fixed now.